oauth2-authorization-server/src/main/java/org/springframework/security/oauth2/server/authorization/web/OAuth2TokenRevocationEndpointFilter.java

```diff
@@ -111,6 +116,12 @@ protected void doFilterInternal(HttpServletRequest request, HttpServletResponse
 			Authentication tokenRevocationAuthentication = this.authenticationConverter.convert(request);
 			Authentication tokenRevocationAuthenticationResult = this.authenticationManager
 				.authenticate(tokenRevocationAuthentication);
+
+			if (tokenRevocationAuthenticationResult instanceof AbstractAuthenticationToken) {
+				((AbstractAuthenticationToken) tokenRevocationAuthenticationResult)
+						.setDetails(this.authenticationDetailsSource.buildDetails(request));
```

Comment:
`setDetails` should be applied on the Authentication Request `tokenRevocationAuthentication` NOT the `tokenRevocationAuthenticationResult`